tests/neg/i2672.scala

```diff
@@ -0,0 +1,16 @@
+class Foo[T]
+
+trait Prepend {
+  type Out
+}
+
+object Test {
+  def foo()(implicit ev: Prepend): Foo[ev.Out] = ???
+
+  def test: Unit = {
+    foo(): Foo[Any] // error: found: Prepend => Foo[_] required: Foo[Any]
```

Comment:
But even if an implicit was found, we'd still get an error because then the function would return a `Foo[_]` where a `Foo[Any]` was expected.